### PR TITLE
Use a custom version of JSON marshalling for license verification

### PR DESCRIPTION
**Pull request by pebrc**
Standard Go json.Marshal escapes "problematic" characters >, <, and & which breaks license verification.

I also generated a test license with external tooling using the key pair we use for unit testing to create a unit test that verifies this works now.

### CODE REVIEW

File: pkg/controller/common/license/verifier.go [hunk 1]
```diff
@@ -229,6 +229,22 @@ func (l EnterpriseLicense) SignableContentBytes() ([]byte, error) {
...
+	encoder := json.NewEncoder(buffer)
+	encoder.SetEscapeHTML(false)
+	err := encoder.Encode(t)
+	if err != nil {
+		return nil, err
+	}
+	marshaledBytes := buffer.Bytes()
+	// json.Encoder adds an additional newline between objects which we do not want here
+	//as it is not part of the signature, so we are removing the last byte
+	return marshaledBytes[:len(marshaledBytes)-1], err
```
Comment (sebgl) [inline pkg/controller/common/license/verifier.go:245]: I'm pretty convinced the slice manipulation will almost never panic (unless empty content in the license?), however I think it'd still be safer to rely on `bytes.TrimSuffix()` here.

File: pkg/controller/common/license/verifier.go [hunk 2]
```diff
@@ -229,6 +229,22 @@ func (l EnterpriseLicense) SignableContentBytes() ([]byte, error) {
...
+	buffer := &bytes.Buffer{}
+	encoder := json.NewEncoder(buffer)
+	encoder.SetEscapeHTML(false)
+	err := encoder.Encode(t)
+	if err != nil {
+		return nil, err
+	}
+	marshaledBytes := buffer.Bytes()
+	// json.Encoder adds an additional newline between objects which we do not want here
+	//as it is not part of the signature, so we are removing the last byte
```
Comment (sebgl) [inline pkg/controller/common/license/verifier.go:244]: nit: missing whitespace after `//`

File: pkg/controller/common/license/verifier.go (hunk 1 above)
Comment (pebrc) [reply pkg/controller/common/license/verifier.go:245]: Ha I had that initially and though it was overkill. But you are right.

**Review by sebgl — APPROVED** [review]
LGTM

**Comment by pebrc** [comment]
Jenkins test this please. 

https://github.com/elastic/cloud-on-k8s/issues/3617 hit again.

**Pull request merged**